docs/UserDefinedLanguageServer.md

```diff
@@ -0,0 +1,64 @@
+# User-defined language server
+
+LSP4IJ provides the capability to consume any language server without developing
+an IntelliJ plugin with `User-defined language server`.
+
+![New Language Server Dialog with TypeScript](./images/user-defined-ls/TypeScriptServerDialog.png)
+
+The main idea is to: 
+
+ * install the language server and its requirements(ex : `NodeJS` to 
+execute a language server written in JavaScript/TypeScript), 
+ * declare the command which starts the language server.
+ * associate the language server with the proper files (identified by IntelliJ Language, File Type or file name pattern)
+
+## New Language Server dialog
+
+To create a new `User-defined language server` you need to open the `New Language Server` dialog:
+
+ * from the menu on right of the LSP console:
+
+![New Language Server From Console](./images/user-defined-ls/NewLanguageServerFromConsole.png)
+
+* with the [+]  on the top of the language server settings:
+
+![New Language Server From Settings](./images/user-defined-ls/NewLanguageServerFromSettings.png)
+
+Once you will click on both of them, the dialog will appear:
+
+![New Language Server Dialog](./images/user-defined-ls/NewLanguageServerDialogEmpty.png)
+
+### Server tab
+
+The server tab requires to fill the `server name` and the `command` which will start the language server.
+
+Here a sample with the [typescript-language-server](https://github.com/typescript-language-server/typescript-language-server):
+
+![New Language Server Dialog with TypeScript](./images/user-defined-ls/TypeScriptServerDialog.png)
+
+### Mappings tab
+
+Mappings tab provides the capability to associate the language server with the proper files identified by 
+
+ * IntelliJ Language, 
+ * File Type language 
+ * File name pattern
+
+Here a sample with the [typescript-language-server](https://github.com/typescript-language-server/typescript-language-server):
+
+![New Language Server Dialog with TypeScript](./images/user-defined-ls/TypeScriptServerDialog.png)
+
+### Configuration tab
+
+TODO
+
+### Debug tab
+
+TODO
+
+### Using template
+
+The Template combox provides some language servers which will prefill server name, command,mappings and configuration. 

```

Comment:
The Template combox provides some language servers templates, pre-filled with server name, command, mappings and potential configuration.